homeassistant/components/mqtt/sensor.py

```diff
@@ -284,12 +284,18 @@ def _update_state(msg: ReceiveMessage) -> None:
             if self.device_class is None:
                 self._attr_native_value = new_value
                 return
-            if (payload_datetime := dt_util.parse_datetime(new_value)) is None:
+            try:
+                if (payload_datetime := dt_util.parse_datetime(new_value)) is None:
+                    _LOGGER.warning(
+                        "Invalid state message '%s' from '%s'", msg.payload, msg.topic
+                    )
+                    return
+            except ValueError:
                 _LOGGER.warning(
                     "Invalid state message '%s' from '%s'", msg.payload, msg.topic
                 )
-                self._attr_native_value = None
```

Comment:
We should not change the native value if there is an error.

Reply:
In all other similar situations we warn but never let the last state become changed.
But if we remove it here it may become a breaking change as it might be actually used.
I'll change it back. If ever needed and we want to remove it we can open a breaking change PR for this.